app/src/main/java/com/stevesoltys/seedvault/settings/SettingsActivity.kt

```diff
@@ -36,6 +37,19 @@ class SettingsActivity : RequireProvisioningActivity(), OnPreferenceStartFragmen
         if (intent?.action == ACTION_APP_STATUS_LIST) {
             showFragment(AppStatusFragment(), true)
         }
+
+        // observe initialization and show/remove init fragment
+        // this can happen when enabling backup and storage wasn't initialized
+        viewModel.initEvent.observeEvent(this) { show ->
+            val tag = "INIT"
+            if (show) {
+                val title = getString(R.string.storage_check_fragment_backup_title)
+                showFragment(StorageCheckFragment.newInstance(title), true, tag)
+            } else {
+                val f = supportFragmentManager.findFragmentByTag(tag)
+                if (f != null && f.isVisible) supportFragmentManager.popBackStack()
```

Comment:
Could be `if(fragment?.isVisible == true)`.

Reply:
Not sure which is more readable, but ok.